pkg/asset/machines/openstack/openstackmachines.go

```diff
@@ -0,0 +1,222 @@
+// Package openstack generates Machine objects for openstack.
+package openstack
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	capo "sigs.k8s.io/cluster-api-provider-openstack/api/v1alpha7"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	machinev1 "github.com/openshift/api/machine/v1"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/manifests/capiutils"
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/openstack"
+)
+
+// GenerateMachines returns manifests and runtime objects to provision the control plane (including bootstrap, if applicable) nodes using CAPI.
+func GenerateMachines(clusterID string, config *types.InstallConfig, pool *types.MachinePool, osImage, role string) ([]*asset.RuntimeFile, error) {
+	if configPlatform := config.Platform.Name(); configPlatform != openstack.Name {
+		return nil, fmt.Errorf("non-OpenStack configuration: %q", configPlatform)
+	}
+	if poolPlatform := pool.Platform.Name(); poolPlatform != openstack.Name {
+		return nil, fmt.Errorf("non-OpenStack machine-pool: %q", poolPlatform)
+	}
+
+	mpool := pool.Platform.OpenStack
+	trunkSupport, err := checkNetworkExtensionAvailability(config.Platform.OpenStack.Cloud, "trunk", nil)
```

Comment:
for reviewers: it's good that we check for trunk availability this way, the install-config option for trunks has been deprecated for a long time.